microservices/workflow/internal/consumer/rabbitmq.go

```
@@ -0,0 +1,481 @@
+		return
+	}
+
+	// Execute the handler
+	err = handler(ctx, message)
+	if err != nil {
+		log.Printf("Error handling message: %v", err)
+
+		// Get current retry count from headers or initialize to 0
+		retryCount := 0
+		if msg.Headers != nil {
+			if count, exists := msg.Headers["x-retry-count"]; exists {
```

golangci-lint (1.64.8)

shadow: declaration of "exists" shadows declaration at line 366

(govet)